src/solvers/dg/2d/dg.jl

```diff
@@ -1879,11 +1877,111 @@ end
 # Calculate and store fluxes across mortars (select correct method based on mortar type)
 calc_mortar_flux!(dg::Dg2D) = calc_mortar_flux!(dg, dg.mortar_type)
 
+# Calculate and store fluxes and nonconservative terms across L2 mortars
+calc_mortar_flux!(dg::Dg2D, mortar_type::Val{:l2}) = calc_mortar_flux!(dg.elements.surface_flux_values, dg, mortar_type,
+                                                                       have_nonconservative_terms(dg.equations), dg.l2mortars, dg.thread_cache)
+function calc_mortar_flux!(surface_flux_values, dg::Dg2D, mortar_type::Val{:l2}, nonconservative_terms::Val{true}, mortars, cache)
+  @unpack neighbor_ids, u_lower, u_upper, orientations = mortars
+  @unpack fstar_upper_threaded, fstar_lower_threaded = cache
+##
+#  FIXME! This is super hacked in for the MHD equations
+##
+  # Type alias only for convenience
+  A2d = MArray{Tuple{nvariables(dg), nnodes(dg)}, Float64}
+
+  noncons_diamond_primary_upper_threaded = [A2d(undef) for _ in 1:Threads.nthreads()]
+  noncons_diamond_primary_lower_threaded = [A2d(undef) for _ in 1:Threads.nthreads()]
+  noncons_diamond_secondary_upper_threaded = [A2d(undef) for _ in 1:Threads.nthreads()]
+  noncons_diamond_secondary_lower_threaded = [A2d(undef) for _ in 1:Threads.nthreads()]
```

Comment:
Should we maybe add a switch to `create_thread_cache_2d` (or via passing an additional argument `have_nonconservative_terms`) to allow creation of these additional caches? @ranocha, what's your take on this?

Reply:
Rightnow, we juat create everything we might need, even if it's not strictly necessary. We can also do that here, I think. It's not too much memory anyway

Reply:
:+1:

Reply:
Postponed to a later point when we do optimizations.